Source/MLXNN/Module.swift

```diff
@@ -1304,7 +1308,15 @@ private protocol TypeErasedSetterProvider {
             module!
         }
         set {
-            module = newValue
+            if module != nil {
```

Comment:
Make ModuleInfo behave the same way as ParameterInfo.  This allows ModuleInfo to be set once if `nil`, e.g. the required set from init.